typing/typecore.ml

```diff
@@ -4585,12 +4621,12 @@ and type_application env funct sargs =
     end
   in
   let warned = ref false in
-  let rec type_args args ty_fun ty_fun0 sargs =
+  let rec type_args args ty_fun ty_fun0 sargs prev_arg_loc =
```

Comment:
Personal nitpicking: as `prev_arg_loc` is a metadata parameter of the functions whereas `sargs` is the decreasing argument in the recursion, I would rather group it with the `args` argument and keep `sargs` as the last argument of the function.

Reply:
I paired the location with the argument, which makes the code clearer. However, I'm not sure it decreases the diff size.